### ISSUE TITLE
Wifi sync flickers in and out - impossible to sync phone to computer

### ISSUE THREAD

**Issue by aliya-ryan** [digidem/mapeo-desktop#276]
Using Mapeo Desktop 5.0.2 and Mapeo Mobile 2.0.0-beta.3

When I try and sync over wifi, phone to computer, the devices (on both phone and computer on sync screens) flicker in and out of connection. I tried this with both mac and windows, and three different phone models. Also tried with internet wifi network and a travel router - they all acted similarly.

Result is that you can not establish a connection long enough to sync phone to computer.

I took a video on my phone but can't seem to get it on here.

**Comment by okdistribute** [comment]
Stack trace: 
```
  discovery-swarm duplicate connections detected in handshake, dropping one +2ms
  mapeo-sync onClose 192.168.0.5 41669 false +2ms
  mapeo-sync emitting sync event 192.168.0.5 41669 false +0ms
  mapeo-sync down { type: 'wifi',
  initiator: true,
  id: 'bce51b375f22',
  host: '192.168.0.5',
  port: 41669,
  channel: <Buffer 34 64 63 64 38 62 66 30 61 36 63 32 65 30 38 64 63 30 65 35 66 66 65 34 30 63 33 30 32 33 39 66 62 35 66 64 39 30 33 37 39 35 36 39 30 66 35 66 36 37 ... >,
  swarmId: <Buffer 9d 98 7d 4b 42 83 c4 60 d9 5d 5d d2 96 70 8a da 52 a0 24 dc c0 8b 33 60 28 62 ab 14 59 67 31 98>,
  connection: 
   Socket {
     connecting: false,
     _hadError: false,
     _handle: null,
     _parent: null,
     _host: null,
     _readableState: 
      ReadableState {
        objectMode: false,
        highWaterMark: 16384,
        buffer: [Object],
        length: 0,
        pipes: [Object],
        pipesCount: 1,
        flowing: true,
        ended: false,
        endEmitted: false,
        reading: true,
        sync: false,
        needReadable: true,
        emittedReadable: false,
        readableListening: true,
        resumeScheduled: false,
        destroyed: true,
        defaultEncoding: 'utf8',
        awaitDrain: 0,
        readingMore: false,
        decoder: null,
        encoding: null },
     readable: false,
     domain: null,
     _events: 
      { end: [Array],
        _socketEnd: [Function: onSocketEnd],
        connect: [Function: onconnect],
        error: [Array],
        close: [Array],
        finish: [Array],
        unpipe: [Function: onunpipe],
        drain: [Function],
        data: [Function: ondata] },
     _eventsCount: 9,
     _maxListeners: undefined,
     _writableState: 
      WritableState {
        objectMode: false,
        highWaterMark: 16384,
        finalCalled: false,
        needDrain: false,
        ending: false,
        ended: false,
        finished: false,
        destroyed: true,
        decodeStrings: false,
        defaultEncoding: 'utf8',
        length: 0,
        writing: false,
        corked: 0,
        sync: false,
        bufferProcessing: false,
        onwrite: [Function: bound onwrite],
        writecb: null,
        writelen: 0,
        bufferedRequest: null,
        lastBufferedRequest: null,
        pendingcb: 0,
        prefinished: false,
        errorEmitted: false,
        bufferedRequestCount: 0,
        corkedRequestsFree: [Object] },
     writable: false,
     allowHalfOpen: false,
     _bytesDispatched: 119,
     _sockname: null,
     _pendingData: null,
     _pendingEncoding: '',
     server: null,
     _server: null,
     _debugId: 13,
     _debugStartTime: 1581037969331,
     _idleNext: null,
     _idlePrev: null,
     _idleTimeout: -1,
     [Symbol(asyncId)]: 2261,
     [Symbol(bytesRead)]: 129 },
  deviceType: 'desktop',
  handshake: { accept: [Function: onaccept] },
  state: { topic: 'replication-wifi-ready', message: undefined },
  sync: 
   EventEmitter {
     domain: null,
     _events: {},
     _eventsCount: 0,
     _maxListeners: undefined },
  name: 'okdistribute' } +0ms
  mapeo-sync pump ended 192.168.0.5 41669 +2ms
  mapeo-sync onClose 192.168.0.5 41669 Error: premature close
    at onclosenexttick (/home/okdistribute/node_modules/mapeo-desktop/node_modules/end-of-stream/index.js:53:86)
    at _combinedTickCallback (internal/process/next_tick.js:132:7)
    at process._tickCallback (internal/process/next_tick.js:181:9) +0ms

```

**Comment by okdistribute** [comment]
Fixed by pinning discovery-swarm to v6.0.0 in mapeo-core https://github.com/digidem/mapeo-core/commits/v8.

Fixed in v5.0.3

**Comment by arky** [comment]
Thank you so much for fixing this issue. Kudos!
